Intersect.Client.Core/Interface/Game/Typewriting/Typewriter.cs

```diff
@@ -57,28 +57,44 @@ public void Write(string? soundName)
         var emitSound = false;
         while (_nextUpdateTime <= Timing.Global.MillisecondsUtc)
         {
-            if (_offset >= _text.Length)
+            if (_segmentIndex >= _segments.Length)
             {
                 End();
-                break;
+                return;
+            }
+
+            var segment = _segments[_segmentIndex];
+
+            if (_offset >= segment.Text.Length)
+            {
+                _offset = 0;
+                _segmentIndex++;
+
+                if (_segmentIndex >= _segments.Length)
+                {
+                    End();
+                    return;
+                }
+
+                segment = _segments[_segmentIndex];
             }
```

Comment:
Move all of this (`var segment = ...` and the if statement) after `emitSound |= ...`